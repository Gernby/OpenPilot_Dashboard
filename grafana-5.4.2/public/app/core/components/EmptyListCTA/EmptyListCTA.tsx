import React, { Component } from 'react';

export interface Props {
  model: any;
}

class EmptyListCTA extends Component<Props, any> {
  render() {
    const {
      title,
      buttonIcon,
      buttonLink,
      buttonTitle,
      onClick,
      proTip,
      proTipLink,
      proTipLinkTitle,
      proTipTarget,
    } = this.props.model;
    return (
      <div className="empty-list-cta">
        <div className="empty-list-cta__title">{title}</div>
        <a onClick={onClick} href={buttonLink} className="empty-list-cta__button btn btn-xlarge btn-success">
          <i className={buttonIcon} />
          {buttonTitle}
        </a>
        <div className="empty-list-cta__pro-tip">
          <i className="fa fa-rocket" /> ProTip: {proTip}
          <a className="text-link empty-list-cta__pro-tip-link" href={proTipLink} target={proTipTarget}>
            {proTipLinkTitle}
          </a>
        </div>
      </div>
    );
  }
}

export default EmptyListCTA;
