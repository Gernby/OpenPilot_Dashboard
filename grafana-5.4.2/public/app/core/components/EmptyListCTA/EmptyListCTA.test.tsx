import React from 'react';
import renderer from 'react-test-renderer';
import EmptyListCTA from './EmptyListCTA';

const model = {
  title: 'Title',
  buttonIcon: 'ga css class',
  buttonLink: 'http://url/to/destination',
  buttonTitle: 'Click me',
  onClick: jest.fn(),
  proTip: 'This is a tip',
  proTipLink: 'http://url/to/tip/destination',
  proTipLinkTitle: 'Learn more',
  proTipTarget: '_blank',
};

describe('EmptyListCTA', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<EmptyListCTA model={model} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
