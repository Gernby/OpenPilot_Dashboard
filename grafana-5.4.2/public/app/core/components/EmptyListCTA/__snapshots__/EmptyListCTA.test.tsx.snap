// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyListCTA renders correctly 1`] = `
<div
  className="empty-list-cta"
>
  <div
    className="empty-list-cta__title"
  >
    Title
  </div>
  <a
    className="empty-list-cta__button btn btn-xlarge btn-success"
    href="http://url/to/destination"
    onClick={[MockFunction]}
  >
    <i
      className="ga css class"
    />
    Click me
  </a>
  <div
    className="empty-list-cta__pro-tip"
  >
    <i
      className="fa fa-rocket"
    />
     ProTip: 
    This is a tip
    <a
      className="text-link empty-list-cta__pro-tip-link"
      href="http://url/to/tip/destination"
      target="_blank"
    >
      Learn more
    </a>
  </div>
</div>
`;
